kubernetes/pod.go

```diff
@@ -38,12 +38,12 @@ func WaitPodReady(ctx context.Context, watcher watch.Interface) (readyChan chan
 				}
 				pod, ok := event.Object.(*corev1.Pod)
 				if !ok {
-					logrus.Fatalf("type assertion failed for object %s, type %T", event.Object.GetObjectKind(), pod)
+					return
```

Comment:
Shouldn't we check if event is nil instead ?

Reply:
Event is not nil when the assertion fails. It means that `event.Object` is a  `*metav1.Status` object